moto/apigateway/models.py

```diff
@@ -1293,7 +1312,11 @@ def create_deployment(
     ) -> Deployment:
         if stage_variables is None:
             stage_variables = {}
-        deployment_id = create_id()
+        # Since there are no unique values to a deployment, we will use the stage name for the deployment.
+        # We are also passing a list of deployment ids to generate to prevent overwriting deployments.
+        deployment_id = ApigwDeploymentIdentifier(
```

Comment:
Having a list of ExistingId's essentially means that it's up to the user to call `moto_id_manager.set_custom_id(..)` before every `Deployment` is instantiated.

So in pseudo code:

```
moto_id_manager.set_custom_id(DeploymentIdentifier, 'suffix')
create_stage(name="mystage")
create_deployment()  # deployment_id = account.region.suffix.mystage
create_deployment()  # deployment_id = random identifier, because 'account.region.suffix.mystage' already exists

moto_id_manager.set_custom_id(DeploymentIdentifier, 'suffix2')
create_deployment()  # deployment_id = account.region.suffix2.mystage
```

Am I understanding the flow correctly?

I can't recall another resource where this is the case, but I'm sure this happens more often.. And it is slightly confusing if the actual behaviour differs for specific Resources.

Do you think it makes sense to implement this in a different way? Maybe make it an incremental generator somehow, so that the names become `account.region.deployment1`, `account.region.deployment2`, etc?